src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx

```diff
@@ -0,0 +1,230 @@
+    let isActiveRouter =
+      !hasPanel &&
+      router &&
+      to &&
+      router.isActive(
+        {
+          pathname: to,
+        },
+        index
+      );
+    let isTop = orientation === 'top';
+    let placement = isTop ? 'bottom' : 'right';
+
+    return (
+      <Tooltip
+        disabled={!collapsed}
+        title={label}
+        tooltipOptions={{placement, html: true}}
+      >
+        <StyledSidebarItem
+          data-test-id={this.props['data-test-id']}
+          active={active || isActiveRouter}
+          href={href}
+          to={to}
+          className={className}
+          onClick={this.handleClick}
+        >
+          <SidebarItemWrapper>
+            <SidebarItemIcon>{icon}</SidebarItemIcon>
+            {!collapsed &&
+              !isTop && (
+                <SidebarItemLabel>
+                  <TextOverflow>{label}</TextOverflow>
+                </SidebarItemLabel>
+              )}
+            {badge > 0 && (
+              <SidebarItemBadge collapsed={collapsed}>{badge}</SidebarItemBadge>
+            )}
+          </SidebarItemWrapper>
+        </StyledSidebarItem>
+      </Tooltip>
+    );
+  }
+}
+
+export default withRouter(SidebarItem);
+
+const getActiveStyle = ({active, theme}) => {
+  if (!active) return '';
+  return css`
+    color: ${theme.white};
+
+    &:focus {
+      color: ${theme.white};
+    }
+
+    &:before {
+      background-color: ${theme.purple};
+    }
+  `;
+};
+
+const StyledSidebarItem = styled(({active, ...props}) => <Link {...props} />)`
+  display: flex;
+  color: inherit;
+  position: relative;
+  cursor: pointer;
+  font-size: 15px;
+  line-height: 32px;
+  height: 34px;
+  flex-shrink: 0;
+
+  transition: 0.15s color linear;
+
+  &:before {
+    display: block;
+    content: '';
+    position: absolute;
+    top: 5px;
+    left: -20px;
+    bottom: 5px;
+    width: 5px;
+    border-radius: 0 3px 3px 0;
+    background-color: transparent;
+    transition: 0.15s background-color linear;
+  }
+
+  @media (max-width: ${p => p.theme.breakpoints[0]}) {
+    margin: 0 4px;
+
+    &:before {
+      top: auto;
+      left: 5px;
+      bottom: -10px;
+      height: 5px;
+      width: auto;
+      right: 5px;
+      border-radius: 3px 3px 0 0;
+    }
+  }
+
+  &:hover,
+  &:focus {
+    color: ${p => p.theme.gray1};
+  }
+
+  ${getActiveStyle};
+`;
+
+const SidebarItemWrapper = styled('div')`
+  display: flex;
+  align-items: center;
+  width: 100%;
+`;
+
+const SidebarItemIcon = styled('span')`
+  content: '';
+  display: inline-block;
+  width: 32px;
+  height: 22px;
+  font-size: 20px;
+
+  svg {
+    display: block;
+    margin: 0 auto;
+  }
+`;
+
+const SidebarItemLabel = styled('span')`
+  margin-left: 12px;
+  white-space: nowrap;
+  opacity: 1;
+  flex: 1;
+`;
+
+const getCollapsedBadgeStyle = ({collapsed, theme}) => {
+  if (!collapsed) return '';
+
+  return css`
+    text-indent: -99999em;
```

Comment:
@ckj do you know if we need this one as well?

Reply:
I feel like there was probably a reason, but it's been like 5 months since I wrote that so I'm not completely sure.

Reply:
Yeah we need this because in the collapsed state, we want to hide the text (badge count).